indico/modules/attachments/controllers/event_package.py

```diff
@@ -41,6 +41,29 @@
 from MaKaC.conference import SubContribution
 
 
+def adjust_path_length(segments):
+    """
+    Shorten the path length to < 260 chars.
+
+    Windows' built-in ZIP tool doesn't like files whose
+    total path exceeds 260 chars. Here we progressively
+    shorten the total until it matches that constraint.
+    """
+    result = []
+    total_len = sum(len(seg) for seg in segments) + len(segments) - 1
+    excess = (total_len - 260) if total_len > 260 else 0
+
+    for seg in reversed(segments):
+        fname, ext = os.path.splitext(seg)
+        cut = min(excess, (len(fname) - 10) if len(fname) > 14 else 0)
+        if cut:
+            excess -= cut
+            fname = fname[:-cut]
+        result.insert(0, fname + ext)
```

Comment:
I'd keep append and returning the reversed version - not that it matters here, but AFAIK inserting at the beginning is more expensive than appending.